core/src/main/java/de/bwaldvogel/mongo/backend/Cursors.java

```diff
@@ -0,0 +1,54 @@
+package de.bwaldvogel.mongo.backend;
+
+import de.bwaldvogel.mongo.backend.aggregation.Aggregation;
+import de.bwaldvogel.mongo.bson.BsonTimestamp;
+import de.bwaldvogel.mongo.bson.Document;
+import de.bwaldvogel.mongo.exception.CursorNotFoundException;
+
+import java.time.Clock;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class Cursors {
+
+    public static final ConcurrentMap<Long, Cursor> cursors = new ConcurrentHashMap<>();
```

Comment:
It is a bad sign to have static state.  There is no way to make this variable instance level ?

Reply:
The alternative is to put it inside the backend and pass it on to the other classes that need to use cursors.

Reply:
yes. it is a better approach.
Avoid static state. All state should be kept in backend instance